en/docs/develop/writing-a-custom-federated-authenticator.md

```diff
@@ -106,1262 +111,118 @@ public interface ApplicationAuthenticator extends Serializable {
-        * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
-        * KIND, either express or implied. See the License for the
-        * specific language governing permissions and limitations
-        * under the License.
-        */
-        package org.wso2.carbon.identity.application.authenticator.social.internal;
-        import org.apache.commons.logging.Log;
-        import org.apache.commons.logging.LogFactory;
-        import org.osgi.service.component.ComponentContext;
-        import org.wso2.carbon.identity.application.authentication.framework.ApplicationAuthenticator;
-        import org.wso2.carbon.identity.application.authenticator.social.facebook2.FacebookCustomAuthenticator;
-        import java.util.Hashtable;
-        /**
-        * @scr.component name="identity.application.authenticator.facebook.component"
-        * immediate="true"
-        */
-        public class FacebookCustomAuthenticatorServiceComponent {
-        private static final Log LOGGER = LogFactory.getLog(FacebookCustomAuthenticatorServiceComponent.class);
-        protected void activate(ComponentContext ctxt) {
-        try {
-        FacebookCustomAuthenticator facebookAuthenticator = new FacebookCustomAuthenticator();
-        Hashtable props = new Hashtable();
-        ctxt.getBundleContext().registerService(ApplicationAuthenticator.class.getName(),
-            facebookAuthenticator, props);
-        if (LOGGER.isDebugEnabled()) {
-            LOGGER.debug("Facebook Custome Authenticator bundle is activated");
-        }
-        } catch (Throwable e) {
-        LOGGER.fatal(" Error while activating Facebook authenticator ", e);
-        }
-        }
-        protected void deactivate(ComponentContext ctxt) {
-        if (LOGGER.isDebugEnabled()) {
-        LOGGER.debug("Facebook Custom Authenticator bundle is deactivated");
-        }
-        }
-        }
-        ```
+1. Start the server and log in to the WSO2 IS Management Console.
 
-### Custom authenticator for Google
+2. Then go to the Service Providers under the main tab. Then add Service Provider name and register it. Let’s use the 
+   playground app and refer [this](https://is.docs.wso2.com/en/latest/learn/deploying-the-sample-app/#deploying-the-playground2-webapp) 
+   to configure playground app.
 
-This section includes the code used to write the custom authenticator
-for Google.
+3. Then List the Service Providers and edit the service provider as follows by navigation to the** OAuth/OpenID Configuration** under **Inbound Authentication Configuration **as explained above.
 
-1.  Download the source code from
-    [here](https://svn.wso2.org/repos/wso2/people/thanuja/org.wso2.carbon.identity.application.authenticator.social/)
-    using the following command on your terminal.
+4. Then click Configure and add[ http://localhost:8080/playground2/oauth2client](http://localhost:8080/playground2/oauth2client) as the call back URL. Click Update.
```

Comment:
```suggestion
4. Click Configure and add `http://localhost:8080/playground2/oauth2client` as the call back URL. Click Update.
```

Reply:
Fixed https://github.com/wso2/docs-is/pull/2527/commits/ffc80dd6ac9088892baa7c56e5f03a9aa0768d3f